### ISSUE TITLE
SignalR.Client in Windows 10 Universal App: System.Net.dll FileNotFoundException

### ISSUE THREAD

**Issue by interware** [SignalR/SignalR#3483]
When calling HubConnection.Start() in Windows 10 Universal App a System.IO.FileNotFoundException is thrown:
Could not load file or assembly 'System.Net, Version=2.0.5.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e, Retargetable=Yes' or one of its dependencies. The system cannot find the file specified.

Stacktrace:
   at Microsoft.AspNet.SignalR.Client.Http.DefaultHttpHandler..ctor(IConnection connection)
   at Microsoft.AspNet.SignalR.Client.Http.DefaultHttpClient.CreateHandler()
   at Microsoft.AspNet.SignalR.Client.Http.DefaultHttpClient.Initialize(IConnection connection)
   at Microsoft.AspNet.SignalR.Client.Transports.TransportHelper.GetNegotiationResponse(IHttpClient httpClient, IConnection connection, String connectionData)
   at Microsoft.AspNet.SignalR.Client.Transports.AutoTransport.GetNegotiateResponse(IConnection connection, String connectionData)
   at Microsoft.AspNet.SignalR.Client.Transports.AutoTransport.Negotiate(IConnection connection, String connectionData)
   at Microsoft.AspNet.SignalR.Client.Connection.Negotiate(IClientTransport transport)
   at Microsoft.AspNet.SignalR.Client.Connection.Start(IClientTransport transport)
   at Microsoft.AspNet.SignalR.Client.Connection.Start()

**Comment by MSiccDev** [comment]
same here ...

**Comment by moozzyk** [comment]
Also reported on StackOverflow: http://stackoverflow.com/questions/29992221/universal-windows-platform-and-signalr-could-not-load-file-or-assembly-system

**Comment by ChrisBriggsy** [comment]
I am having the same issue

**Comment by interware** [comment]
Vanman314, I hear where you are coming from. The frustration of having all the experience but being denied from truly contributing to a topic. I for myself am also denied the right to comment on StackOverflow, because I lack reputation. . It's even more stupid when you see people having that right but at the same time completely lacking any kind of competence.
This all adds to my discomfort with GitHub and OpenSource in general.
And still did I jump over my shadow to come here and help get some attention to the problem that is bothering you too (not anymore it seems).

**Comment by zeroskyx** [comment]
Any news on the issue?
vanman314, please share your answer with us.
Thanks in advance!

**Comment by MSiccDev** [comment]
would also love to see @vanman314's solution to this...

**Comment by dsplaisted** [comment]
A workaround for this should be to compile the SignalR client DLL without support for SL5 - targeting .NET 4.5, Windows 8, Windows Phone Silverlight 8, and Windows Phone 8.1, also known as PCL Profile 259.

**Comment by MSiccDev** [comment]
@dsplaisted just tried your suggestion, but removing SL5 means no longer being able to compile over all. There are too many dependencies that get broken.

**Comment by moozzyk** [comment]
@weshaggard suggested you could compile in release as a workaround. While this seems to solve the issue with credentials I was still not able to connect to the server due to: 

```
Method 'Thread.get_CurrentUICulture()' was not included in compilation. There may have been a missing assembly.

   at Microsoft.AspNet.SignalR.TaskAsyncHelper.TaskRunners`2.RunTask(Task`1 task, Func`2 successor)
   at Microsoft.AspNet.SignalR.TaskAsyncHelper.Then[T,TResult](Task`1 task, Func`2 successor)
   at Microsoft.AspNet.SignalR.Client.Http.DefaultHttpClient.Get(String url, Action`1 prepareRequest, Boolean isLongRunning)
   at Microsoft.AspNet.SignalR.Client.Transports.TransportHelper.GetNegotiationResponse(IHttpClient httpClient, IConnection connection, String connectionData)
   at Microsoft.AspNet.SignalR.Client.Transports.AutoTransport.Negotiate(IConnection connection, String connectionData)
   at Microsoft.AspNet.SignalR.Client.Connection.Start(IClientTransport transport)
   at App3.MainPage.<button_Click>d__1.MoveNext() in C:\Users\pawelka\Documents\Visual Studio 2015\Projects\App3\App3\MainPage.xaml.cs:line 34
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() in f:\dd\ndp\fxcore\src\System.Private.CoreLib\System\Runtime\ExceptionServices\ExceptionDispatchInfo.cs:line 66
   at System.Threading.Tasks.ExceptionDispatchHelper.<>c__DisplayClass0.<ThrowAsync>b__1(Object edi) in f:\dd\ndp\fxcore\src\System.Private.Threading\System\Runtime\CompilerServices\AsyncMethodBuilder.cs:line 937
   at System.Threading.WinRTSynchronizationContext.Invoker.InvokeCore() in f:\dd\ndp\fxcore\src\System.Private.Threading\System\Threading\SynchronizationContext.cs:line 170
```

It looks like there will be some work on the SignalR side required to enable the client on Win10 UAP.

**Comment by weshaggard** [comment]
Unfortunately in RC timeframe Thread.CurrentUICulture supported was part of .NET Native (i.e. the difference between debug vs release in Win10 UWP app). I expect that both these issues to be addressed for RTM.

**Comment by LocalJoost** [comment]
Hi, it's a bit crude, but I have found a workaround. Please see this short write-up http://dotnetbyexample.blogspot.com/2015/05/getting-signalr-clients-to-work-on.html

**Comment by zeroskyx** [comment]
That works flawlessly LocalJoost!
I'm not very familiar with the process around here -- how can we get this fix into the official code?

**Comment by LocalJoost** [comment]
Ehm, I don't quite know. Is there already an official NuGet moniker for UWP apps? That would be the way forward I think.

Joost van Schaik

Windows Platform Development MVP
Sent from my Lumia 1520 running Windows Phone 8.1 update 1

"The world as I envision it in my head is such a more interesting place"
________________________________

**Comment by bc3tech** [comment]
related to https://github.com/SignalR/SignalR/issues/3547?

**Comment by bc3tech** [comment]
@MSiccDev found this today as well: http://dotnetbyexample.blogspot.com/2015/05/getting-signalr-clients-to-work-on.html

**Comment by moozzyk** [comment]
Fixed in 4a46099e9323d3be250bc1f013421ffd10a96397 9e5449bdff81b9f85a9605739651db5d14234c5f be8a4ed8238ec8f5f08fd5fb04a14e35d7ad4145 edeacf4cbadffda58e65452687b793b4a1fea9f5
